editor/inserter/test/menu.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * External dependencies
+ */
+import { shallow } from 'enzyme';
+import enzymeToJson from 'enzyme-to-json';
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { registerBlockType, unregisterBlockType, getBlockTypes } from 'blocks';
+
+/**
+ * Internal dependencies
+ */
+import { InserterMenu } from '../menu';
+
+const textBlock = {
+	name: 'core/text-block',
+	title: 'Text',
+	save: noop,
+	edit: noop,
+	category: 'common',
+};
+
+const advancedTextBlock = {
+	name: 'core/advanced-text-block',
+	title: 'Advanced Text',
+	save: noop,
+	edit: noop,
+	category: 'common',
+};
+
+const someOtherBlock = {
+	name: 'core/some-other-block',
+	title: 'Some Other Block',
+	save: noop,
+	edit: noop,
+	category: 'common',
+};
+
+const moreBlock = {
+	name: 'core/more-block',
+	title: 'More',
+	save: noop,
+	edit: noop,
+	category: 'layout',
+	useOnce: 'true',
+};
+
+const youtubeBlock = {
+	name: 'core/youtube-block',
+	title: 'Youtube',
+	save: noop,
+	edit: noop,
+	category: 'embed',
+};
+
+describe( 'InserterMenu', () => {
+	const unregisterAllBlocks = () => {
+		getBlockTypes().forEach( ( block ) => {
+			unregisterBlockType( block.name );
+		} );
+	};
+
+	afterEach( () => {
+		unregisterAllBlocks();
+	} );
+
+	beforeEach( () => {
+		unregisterAllBlocks();
+		registerBlockType( textBlock.name, textBlock );
+		registerBlockType( advancedTextBlock.name, advancedTextBlock );
+		registerBlockType( someOtherBlock.name, someOtherBlock );
+		registerBlockType( moreBlock.name, moreBlock );
+		registerBlockType( youtubeBlock.name, youtubeBlock );
+	} );
+
+	it( 'should show the recent tab by default', () => {
+		const wrapper = shallow(
+			<InserterMenu
+				instanceId={ 1 }
+				blocks={ [] }
+				recentlyUsedBlocks={ [] }
+			/>
+		);
+
+		expect( enzymeToJson( wrapper ) ).toMatchSnapshot();
```

Comment:
I don't like these assertions personally, they break on each small change to the component. I'd rather add more targeted tests